lib/init.spec.lua

```diff
@@ -264,4 +264,61 @@ return function()
 			expect(#chained._values).to.equal(0)
 		end)
 	end)
+
+	describe("Promise.all", function()
+		it("should error if given something other than a table", function()
+			expect(function()
+				Promise.all(1)
+			end).to.throw()
+		end)
+
+		it("should resolve instantly with an empty table if given no values", function()
+			local promise = Promise.all({})
+			local success, value = promise:await()
+
+			expect(success).to.equal(true)
+			expect(promise:getStatus()).to.equal(Promise.Status.Resolved)
+			expect(value).to.be.a("table")
+			expect(#value).to.equal(0)
+		end)
+
+		it("should let non-promise values pass through unchanged", function()
+			local values = {{}, {}, {}}
+			local promise = Promise.all(values)
+			local success, resolved = promise:await()
```

Comment:
Do we want to support non-promises in `Promise.all`? That seems like a mistake to me -- it's trivial to turn a primitive value into a promise with `Promise.resolve(x)`, so expecting that all inputs to `Promise.all` are promises seems reasonable to me.

Reply:
The reason I implemented it this way is because JS's Promise.all just passes non-promise values through as well. I figured that it'd be best to parallel that implementation. Something has *probably* gone wrong if non-promises are being passed into Promise.all, but there may also be value in having this flexibility.

I imagine this might be helpful in some situation where a function expects an array of promises, but for some reason the developer wants to invoke the function with predefined values. In that case, it might be useful to pass the array of values through Promise.all to get promise wrappers around them.

Seeing as JavaScript's Promise.all just wraps non-promise values in a promise, I think it'd be best to mirror that here. They probably put thought into it (at least I'd hope so) :P

Reply:
This might be a case that's worth diverging from JS. Passing a non-promise into `Promise.all` seems like something that'll certainly bite people!

Reply:
Probably for the best. Changed to throw if given non-promises.